src/System.Private.CoreLib/shared/System/Globalization/TimeSpanParse.cs

```diff
@@ -1690,7 +1691,7 @@ private static bool TryParseExactMultipleTimeSpan(ReadOnlySpan<char> input, stri
             // one of the formats.
             for (int i = 0; i < formats.Length; i++)
             {
-                if (formats[i] == null || formats[i].Length == 0)
+                if (formats[i] == null || formats[i]!.Length == 0)
```

Comment:
Why is this ! needed?  I would expect that the compiler would see that this isn't null, as we just checked (and it only cares about local modifications, such that it wouldn't be concerned about some other thread changing the array).

Reply:
seems like a compiler bug to me - I get compile error when I remove `!`. cc: @jaredpar

Reply:
Maybe add TODO-NULLABLE here?